File: https:/github.com/rajesh-kayal-dev/Leetcode.git/11-ContainerWithMostWater/11-ContainerWithMostWater.java
// Last updated: 6/15/2026, 10:03:37 AM
class Solution {
    public int maxArea(int[] height) {
        int left =0;
        int right = height.length -1;
        int maxwater=0;

        while(left < right){
            int minHeight = Math.min(height[left], height[right]);
            int width = right - left;
            int area = minHeight * width;
            maxwater= Math.max(maxwater, area);

            if(height[left] < height[right]) left++;
            else right--;
        }

        return maxwater;
    }
}

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/11-ContainerWithMostWater/11-ContainerWithMostWater.js
// Last updated: 6/15/2026, 10:03:42 AM
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left = 0;
    let right = height.length - 1;
    let maxWater = 0;

    while(left < right){
        let currentHeight = Math.min(height[left], height[right]);
        let currentWidth = right - left;
        let currentArea = currentWidth * currentHeight;
        maxWater = Math.max(maxWater, currentArea);

        if(height[left] < height[right]) left++;
        else right--;
    }
    return maxWater;
};